### PR TITLE
PT-159405636 Create channel contract from on chain

### PR DESCRIPTION
**Pull request by cytadela8**
Documentation in aeternity/protocol#327

### CODE REVIEW

**Comment by cytadela8** [comment]
I understand this conflicts with fortuna and it's blocked by it.

**Comment by lucafavatella** [comment]
I am taking the liberty of labelling as WIP as not planned for this release - so to ease filtering of open PRs.

Also seems requiring rebase.

File: apps/aehttp/src/sc_ws_api_jsonrpc.erl [hunk 1]
```diff
@@ -258,6 +258,28 @@ process_request(#{<<"method">> := <<"channels.update.new_contract">>,
...
+        {{ok, OnChainContract}, {ok, CallData}} ->
+            case aec_chain:get_contract(OnChainContract) of
+                {ok, Contract} ->
+                    case aesc_fsm:upd_create_contract(FsmPid,
+                        #{vm_version    => aect_contracts:vm_version(Contract),
+                            abi_version => aect_contracts:abi_version(Contract),
+                            deposit     => Deposit,
+                            code        => aect_contracts:code(Contract),
+                            call_data   => CallData}) of
+                        ok -> no_reply;
```
Comment (lucafavatella) [inline apps/aehttp/src/sc_ws_api_jsonrpc.erl]: I see this follows `channels.update.new_contract`. Ok.

File: apps/aehttp/test/aehttp_integration_SUITE.erl [hunk 2]
```diff
@@ -4231,15 +4231,62 @@ sc_ws_contract_generic_(Origin, Fun, Config, Opts) ->
...
-    [Fun(Owner, GetVolley, SenderConnPid,
+    [Fun(Owner, GetVolley, CreateContract, SenderConnPid,
          AckConnPid, OwnerPubkey, OtherPubkey, Opts, Config)
         || {Owner, {OwnerPubkey, OtherPubkey}} <- Actors],
     ok = ?WS:unregister_test_for_channel_events(SenderConnPid, [sign, info, get, error]),
     ok = ?WS:unregister_test_for_channel_events(AckConnPid, [sign, info, get, error]),
     ok.
 
-sc_ws_oracle_contract_(Owner, GetVolley, ConnPid1, ConnPid2,
+post_contract_onchain(EncodedCode, EncodedInitCallData, Deposit) ->
```
Comment (lucafavatella) [inline apps/aehttp/test/aehttp_integration_SUITE.erl]: It seems this function is unused in this commit. Though ok.

File: apps/aehttp/src/sc_ws_api_jsonrpc.erl [hunk 3]
```diff
@@ -258,6 +258,28 @@ process_request(#{<<"method">> := <<"channels.update.new_contract">>,
             end;
         _ -> {error, broken_code}
     end;
+process_request(#{<<"method">> := <<"channels.update.new_contract.from_onchain">>,
```
Comment (lucafavatella) [inline apps/aehttp/src/sc_ws_api_jsonrpc.erl]: I see `.` changed to `_` in following commit. Ok.

**Pull request merged**